### ISSUE TITLE
[BUG]: Inconsistent Sprite Scaling when rendering at 2x+ and being processed by the FRU. Results are off by 1.

### ISSUE THREAD

**Issue by alexankitty** [PCSX2/pcsx2#8145]
### Describe the Bug

In the game Shin Megami Tensei Persona 3 FES there is an issue with character portraits (bust ups) where the blink texture displayed over the portrait is misaligned. Upon further inspection it seems to be a problem with how the sprites are scaled. I've created a pnach to mitigate this issue and display the assembly code at question here. I believe there is a floating point rounding error that's occurring during the sprite scaling. Supporting evidence included under logs.

### Reproduction Steps

* Open Persona 3 FES
* Use any renderer except for software
* Set rendering resolution to 2x or higher
* Talk to any character in the game with and have them display a character portrait
* Wait for the blink effect to see how half of the blink effect is in the wrong place.

### Expected Behavior

The blink effect should be correctly stretched over the character portrait and should be mostly seamless.

### PCSX2 Revision

v1.7.4073

### Operating System

Windows 11

### CPU

Intel Core i9 10900K

### GPU

Nvidia Geforce RTX 3090

### GS Settings

![image](https://user-images.githubusercontent.com/45508320/218617096-b438abe4-c30e-4ab4-be76-f6d899a2302b.png)
![image](https://user-images.githubusercontent.com/45508320/218617118-7763d6d0-e855-46a2-9f2b-46b182d6e037.png)

### GS Window Screenshots

Have to use a video file to demonstrate the issue as it's hard to convey soley via screenshots.
https://user-images.githubusercontent.com/45508320/218618770-ea55e310-4ecf-42da-822c-03f20e722f36.mp4

### Logs & Dumps

Without PNACH mitigation
[SLUS_216.21_SLUS-21621_20230213185131.gs.zip](https://github.com/PCSX2/pcsx2/files/10727864/SLUS_216.21_SLUS-21621_20230213185131.gs.zip)
With PNACH mitigation
[SLUS_216.21_SLUS-21621_20230213185310.gs.zip](https://github.com/PCSX2/pcsx2/files/10727870/SLUS_216.21_SLUS-21621_20230213185310.gs.zip)

Here's the pnach contents to work around the issue
`//Upscale Blink fixes (Alexankitty)
//patch=1,EE,0010621C,word,3C024282 //stretch the blink by 2.
//patch=1,EE,0010624C,word,3C024201 //move the blink position down by 1`
The issue is still present even with these, so it seems like somewhere something is off by 1.
The memory addresses that control this output are listed in the pnach.
I've spent a few hours trying to come up with a fix for this and it doesn't matter how I manipulate the values, it's always just barely off.

**Comment by refractionpcsx2** [comment]
Yep, this is a well known issue, which is something we have to work on.  

For example we will have a 16x16 texture, and if you draw all of that, you would do 0,0 -> 15, 15.  In native, you get that, but when upscaled to say x2 you get 0,0 -> 30, 30, not 31, 31.  This is certainly something we have to fix, but it's probably going to be a hefty job.

**Comment by alexankitty** [comment]
Understood. Hopefully what I've provided is at least somewhat useful then for tackling this.

**Comment by destroyer449** [comment]
> Yep, this is a well known issue, which is something we have to work on.
> 
> For example we will have a 16x16 texture, and if you draw all of that, you would do 0,0 -> 15, 15. In native, you get that, but when upscaled to say x2 you get 0,0 -> 30, 30, not 31, 31. This is certainly something we have to fix, but it's probably going to be a hefty job.

Correct me if I'm wrong, but couldn't this be fixed by calculating the end index as `(original index + 1)*resolution -1` instead? This would cause 15 to correctly go to 31 or whatever higher resolution end point is needed.

**Comment by refractionpcsx2** [comment]
it's not just sprites, is the size of triangles too, and that's a whole nightmare kettle of fish.
